docs/389ds/design/accesslog-json-design.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Access Log JSON Format Design"
+---
+
+# Access Log JSON Format
+----------------
+
+Overview
+--------
+
+In order to make the access logging more consumable by standard parsing tools there should be an option for writing the log in JSON
+
+JSON Design
+------------
+
+    {
+        // Header
+        version: "",
+        instance: "",
+        // The rest of the log
+        op: (
+            ABANDON, ADD, AUTOBIND, BIND/UNBIND, CLOSE, CMP, CONN, DEL, EXT,
+            MOD, MODRDN, SEARCH, SORT, STAT, RESULT, VLV
+        ),
+        local_time: <strftime output - customizable>,
+        gm_time: <gm time - uses a fixed format of **%FT%TZ**>,
+        conn_id: ####,
+        op_id: ####,
```

Comment:
I'm not sure if it's already here, but I think we need to have a unique identifier for each log entry to help with deduplication and tracking. Probably by combining a few fields.

Reply:
Do you mean a single header?  Or a unique value in every single log event?  You could argue the timestamp is unique.  I guess I'm not sure what you are trying to solve here

Reply:
I think timestamp can be the same (but very rarely) on some concurrent events.

For any log consumption of the most log processing tools, it's beneficial to distinguish between events with just one field - uniq key.
It's a safe and future proof option to add one, IMO...

Reply:
I agree, I have recently noticed that opid+connid is not sufficient to identify all event associated with a given operation (as different connection may have the same connid) we probably need a 64 bit counter or associate the operation start timestamp + connid + opid anyway   having an unique operation identifier will be helpfull

Reply:
And we may use the same  field /object for the response control in RESULT record

Reply:
> I agree, I have recently noticed that opid+connid is not sufficient to identify all event associated with a given operation (as different connection may have the same connid) we probably need a 64 bit counter or associate the operation start timestamp + connid + opid anyway having an unique operation identifier will be helpfull

Sounds good, I'll add some kind of unique key for each connection and its ops.

Reply:
What about log rotation. If an operation/connection starts on a file and ends on another file. Do we need to match the unique key between files ?
Also, if an admin wants to concatenate several log files, will it break the unique key logic ?

Reply:
> What about log rotation. If an operation/connection starts on a file and ends on another file. Do we need to match the unique key between files ? Also, if an admin wants to concatenate several log files, will it break the unique key logic ?

That all happens behind the scenes.  The unique id will not be impacted by log rotation - it will remain the same throughout the life of the connection.